utils/datetime/LocalDate.js

```diff
@@ -100,35 +101,50 @@ export class LocalDate {
     //--------------------
     // Getters and basic methods.
     //--------------------
+    /** @return {string} */
     get isoString() {
         return this._isoString;
     }
 
+    /** @return {Date} */
     get date() {
         return new Date(this.timestamp);
     }
 
+    /** @return {moment} */
     get moment() {
         return this._moment.clone();
     }
 
+    /** @return {number} */
     get timestamp() {
         return this._date.getTime();
     }
 
+    /** @return {string} */
     format(...args) {
         return this._moment.format(...args);
     }
 
+    /** @return {string} */
     dayOfWeek() {
```

Comment:
This little pattern seems like a poster child for an annotation honestly -- would be called something like "@lazy" or "@computeOnce".  It could even store the property on the object itself as ._lazyPropName if that was considered more efficient or straightforward (not sure that it is)

Reply:
And reading more carefully your comment, see that you mention that.

Reply:
am curious to know what prompted the caching bit -- are we doing some of these operations in a renderer, or otherwise where it feels like we are doing it in bulk wastefully?

Certainly we could easily add a decorator for this -- the question is how to generalize, it and choose an implementation.  Could see it becoming more important in other classes, where perhaps it also takes into account the parameters to the function (see Lodash memoize) just want to get this right, but also not hold up the unrelated changes.  

Also we might as well just add endOfYear/startOfYear while we are at it.   Happy to make these changes.